### PR TITLE
Chapter12

### PR DESCRIPTION
**Pull request by NurAnisAqilah**

### CODE REVIEW

**Comment by NurAnisAqilah** [comment]
ready

File: app/assets/stylesheets/password_resets.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the PasswordResets controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/password_resets.scss]: remove file that never used

File: app/helpers/password_resets_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module PasswordResetsHelper
+end
```
Comment (saobv-2223) [inline app/helpers/password_resets_helper.rb]: remove file that never used

File: app/mailers/user_mailer.rb [hunk 3]
```diff
@@ -20,8 +25,8 @@ def account_activation(user)
     mail to: user.email, subject: "Account activation"
   end
 
-  def password_reset
-    @greeting = "Hi"
-    mail to: "to@example.org"
+  def password_reset(user)
```
Comment (saobv-2223) [inline app/mailers/user_mailer.rb]: remove ()

File: app/views/password_resets/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<h1>PasswordResets#edit</h1>
+<p>Find me in app/views/password_resets/edit.html.erb</p>
+
+<% provide(:title, "Reset password") %>
+<h1>Reset password</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user, url: password_reset_path(params[:id])) do |f| %>
```
Comment (saobv-2223) [inline app/views/password_resets/edit.html.erb]: you can remove ()

File: config/routes.rb [hunk 5]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'password_resets/new'
+  get 'password_resets/edit'
```
Comment (saobv-2223) [inline config/routes.rb]: using "

File: config/routes.rb [hunk 6]
```diff
@@ -10,6 +12,7 @@
 
   resources :users, only: %i(show new)
   resources :account_activations, only: :edit
+  resources :password_resets, only: [:new, :create, :edit, :update]
```
Comment (saobv-2223) [inline config/routes.rb]:
```suggestion
  resources :password_resets, except: %i(index show destroy)
```

**Comment by NurAnisAqilah** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**